lib/features/scanner/scanner_bloc.dart

```diff
@@ -0,0 +1,185 @@
+
+import 'package:barcode_scan/barcode_scan.dart';
```

Comment:
We said a block is a stream of a state - each stream update describes a new state

What defines the state in this bloc?

Reply:
I view the BLOC as the class representing the state in this case. 

I would say that the UI can get a snapshot of the current state from the BLOC via it's API. I.e. the public Stream<?> "getters". 

If you look inside the BLOC the stateful "variables" are the BehaviorSubjects, but the interaction between those are just as important in most cases. 

```
  final _available = BehaviorSubject<ScannerAvailable>();
  final _scanResult = BehaviorSubject<ScanResult>();
  final _status = BehaviorSubject<ScanStatus>();
  final _execute = PublishSubject<ScannerCmd>();
  final _textController = BehaviorSubject<SetTextControllerCmd>();
  final _inviteCode = BehaviorSubject<String>();
```

The _execute stream doesn't keep anything and is my own (I've not seen anyone else do that) way of altering the BLOC state inspired by CQRS (https://en.wikipedia.org/wiki/Command%E2%80%93query_separation). That's why all of the state changing DTOs are named Commands.